process/block/metablock.go

```diff
@@ -827,6 +826,11 @@ func (mp *metaProcessor) receivedShardHeader(shardHeaderHash []byte) {
 		mp.hdrsForCurrBlock.mutHdrsForBlock.Unlock()
 	}
 
+	if mp.isHeaderOutOfRange(shardHeader, shardHeaderPool) {
```

Comment:
this might result in errors. LIke mp.chRcvAllHdrs gets set to true, then you forcefully deleted a shardHeader.

Reply:
The header will be deleted only if it is far away from lastNotarized, like 90% from pool size away in the term of nonces. This is the case when you are in bootstrap mode and you are receiving headers from the network with higher nonces than what you are processing now. This mechanism tries to reserve at least 10% from pool for current nonces which have been requested as part of the current block which needs to be processed.